### ISSUE TITLE
Changing Linux Config Parameter

### ISSUE THREAD

**Issue by Christian56** [OP-TEE/optee_os#2559]
Hi, 
I'm trying to add config parameters to the .config file. I tried to add these Parameter to 
/build/kconfigs/rpi3.conf and /linux/arch/arm64/configs/bcmrpi3_defconfig. However when i compile the system these Parameters are not in the config file. 
When I use the make linux-defconfig command i get the following output: 

```
Value requested for CONFIG_IMA not in final .config 
Requested value:  CONFIG_IMA=y 
Actual value:     
```
So I I think that i might use the wrong config files, but I'm not sure which one to use.
I also tried to add the parameters into .config and just make the kernel without building the system but it seems to always generate a new .config file.

**Comment by jforissier** [comment]
Hi @Christian56,

You probably need to enable some other config flag that CONFIG_IMA depends on.

**Comment by Christian56** [comment]
Hi @jforissier,

 based on https://sourceforge.net/p/linux-ima/wiki/Home/#compiling-the-kernel-with-evmima-appraisal-enabled I  just added the parameters necessary for IMA: 
```
CONFIG_INTEGRITY=y 
CONFIG_IMA=y 
CONFIG_IMA_MEASURE_PCR_IDX=10 
CONFIG_IMA_AUDIT=y 
CONFIG_IMA_LSM_RULES=y 
```
But still got the same message.

**Comment by Emantor** [comment]
This may be the valid configuration for an older kernel version, for the kernel configuration it's usually best to activate the configuration option via `make menuconfig`or similar commands. That way the kernel configuration system will select the correct dependencies itself.

**Comment by Christian56** [comment]
Ah thanks for the hint @Emantor. With `make menuconfig` I got to compile the kernel including IMA. However it freezes after the following message:
```
INFO:    BL3-1: Preparing for EL3 exit to normal world
INFO:    BL3-1: Next image address = 0x80000
INFO:    BL3-1: Next image spsr = 0x3c9
```
What I did was to load the normal config activate IMA and removed  "Rasperry Pi display firmware driven backlight" from device drivers/graphic support / backlight & lcd device support as I got an error during compilation otherwise and it didnt seem to important to me. After that i compiled used the default config options when being asked. 

So I assume i need to change other config parameters in order to successfully compile the kernel. But I don't know what Configuration.
